pkgs/top-level/octave-packages.nix

```diff
@@ -0,0 +1,91 @@
+# This file contains the GNU Octave add-on packages set.
+# Each attribute is an Octave library.
+# Expressions for the Octave libraries are supposed to be in `pkgs/development/octave-modules/<name>/default.nix`.
+
+{ pkgs
+, lib
+, stdenv
+, fetchurl
+, newScope
+, octave
+, lapack, blas
+, gfortran
+# Use the same python that octave is using
+, python
+# Use the same Java that octave is using
+, enableJava
+, jdk
+, gnuplot
+, texinfo
+, nettle
+}:
+
+let
+
+  isOctaveFull = octave.enableQt;
+  isOctaveJIT = octave.enableJIT;
+
+  buildOctaveLibrary = callPackage ../development/interpreters/octave/mk-octave-derivation.nix {
+    inherit pkgs lib stdenv;
+    namePrefix = "${octave.pname}-${octave.version}";
+    inherit octave;
+  };
+
+  callPackage = pkgs.newScope {
+    inherit (pkgs) lib stdenv;
+    inherit buildOctaveLibrary;
+    inherit fetchurl;
+    inherit gnuplot texinfo;
```

Comment:
Even if not using `lib.makeScope`, this is where other deps from `octave` should be inherited.

Reply:
See https://github.com/NixOS/nixpkgs/pull/108562/commits/b58f29416eee44575e039d463c7cb665555416e9